tiledb/libtiledb.pyx

```diff
@@ -2543,9 +2543,18 @@ def index_domain_subarray(array: Array, dom: Domain, idx: tuple):
 
         if np.issubdtype(dim_dtype, np.unicode_) or np.issubdtype(dim_dtype, np.bytes_):
             ned = array.nonempty_domain()
-            (dim_lb, dim_ub) = ned[r] if ned else (None, None)
+            (dim_lb, dim_ub) = ned[r] if ned is not None else (None, None)
         else:
-            (dim_lb, dim_ub) = dim.domain
+            if read_array:
+                with Array.load_typed(array.uri) as A:
```

Comment:
I'm not quite following why this is done here, but we need to avoid re-opening the array inline if at all possible... There are many events that could violate expectations here (eg if the array has been updated since the parent-array caller's timestamp, which might change the fragments, nonempty domain, etc.).

Reply:
Is there a way to get the non-empty domain without opening the array? When I tried doing `array.nonempty_domain()` it complained "Array is not open for reading."